geniesp/bpc_redcap_export_mapping.py

```diff
@@ -235,50 +313,22 @@ def get_drug_mapping(
         cohort (str): cohort label
         synid_file_grs (str): Synapse ID of REDCap global response set file.
         synid_table_prissmm (str): Synapse ID of PRISSMM documentation table
+        use_grs (bool): Whether to use GRS or not
 
     Returns:
-        dict: map where keys are BPC drug short names and value is the
+        Dict[str, str]: map where keys are BPC drug short names and value is the
                 corresponding NCIT drug code
     """
-
-    mapping = {}
-    var_names = []
-
-    # ! Not removing data dictionary code until 100% certain
-    # synid_file_dd = _get_synid_dd(syn, cohort, synid_table_prissmm)
-
-    # dd = pd.read_csv(
-    #     syn.get(synid_file_dd).path, encoding="unicode_escape", low_memory=False
-    # )
-    grs = pd.read_csv(
-        syn.get(synid_file_grs).path, encoding="unicode_escape", low_memory=False
+    synid_file_dd = _get_synid_dd(syn, cohort, synid_table_prissmm)
+    mapping = get_mapping_data(
```

Comment:
@rxu17 we may need to revisit this logic, but thats on me.

So before this GRS deprecation flip flop, the for loop below looked something like this:

```
for obj in [dd, grs]:
```

Which means that the dd was also used and then GRS was used to replace the other values (similar to how it's being done in the genie-bpc-pipeline.

Reply:
Yes i think it's cohort dependent from previous discussions? If that is the case I think maybe we have to instead add `use_grs` to the config for the cohort dependent parameters instead of offering it as a parameter in the launch....

Reply:
So we _could_ add it to the config but that would be out of scope for this ticket for now - we can ask chelsea.

What I was more pointing out is currently your code uses DD or GRS (I think), but when we have the use_grs flag, we need to pull values from DD and then do a full outer join with values from GRS.

If there are values in GRS that overwrite those in DD, GRS takes precedence.

Reply:
Ah so what you said here: https://sagebionetworks.slack.com/archives/C0350GLJ7M4/p1730926936700789?thread_ts=1729275644.884449&cid=C0350GLJ7M4

But based on [this](https://sagebionetworks.slack.com/archives/C0350GLJ7M4/p1730925291270629?thread_ts=1729275644.884449&cid=C0350GLJ7M4) and previous statements, my understanding was we don't have an available fix so we use DD instead of DD+GRS and we would only be able to use GRS alone or DD hence the above decision. If this isn't the case, we should definitely chat about it.

Reply:
Ah. I stand corrected - let's go with what Chelsea said.

I do agree that doing a union probably isn't great as it mixes data dictionary versions.  Thanks!